tools/cuttlefish-host-image-installer/addpreseed_aosp.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/sh
+
+BASEDIR=$(dirname $(realpath "$0"))
+
+orig_iso=mini.iso
+auto_extract_efi=1
+efi_img=efi.img
+
+new_files=iso_unpacked_and_modified
+new_iso=preseed-mini_aosp.iso
+
+PRESEEDFILE=$(realpath "${BASEDIR}"/preseed/preseed.cfg)
+AFTERINSTALLSCRIPT=$(realpath "${BASEDIR}"/preseed/after_install_1_aosp.sh)
+
+part_img_ready=1
+if test "$auto_extract_efi" = 1; then
+  start_block=$(/sbin/fdisk -l "$orig_iso" | fgrep "$orig_iso"2 | \
+                awk '{print $2}')
+  block_count=$(/sbin/fdisk -l "$orig_iso" | fgrep "$orig_iso"2 | \
+                awk '{print $4}')
+  if test "$start_block" -gt 0 -a "$block_count" -gt 0 2>/dev/null
+  then
+    dd if="$orig_iso" bs=512 skip="$start_block" count="$block_count" \
+       of="$efi_img"
+  else
+    echo "Cannot read plausible start block and block count from fdisk" >&2
+    part_img_ready=0
+  fi
+fi
+
+# add preseed
+mkdir ${new_files}
+bsdtar -C ${new_files} -xf "$orig_iso"
+cd ${new_files}
+cp -f "${PRESEEDFILE}" preseed.cfg
+cp -f "${AFTERINSTALLSCRIPT}" after_install_1.sh
+chmod a+rx after_install_1.sh
+
+# add preseed to console based installer
+chmod ug+w initrd.gz
+gzip -d -f initrd.gz
+echo preseed.cfg | cpio -H newc -o -A -F initrd
+echo after_install_1.sh | cpio -H newc -o -A -F initrd
+gzip -9 initrd
+chmod a-w initrd.gz
+# add preseed to GTK based installer
+chmod ug+w gtk
+cd gtk
```

Comment:
nit:
```
cd somedir
# do stuff in somedir
cd ..
```

is better handled using `pushd`:
```
pushd somedir
# do stuff in somedir
popd
```
If `somedir` is ever moved to location deeper in the directory tree, just modifying the `pushd` line is enough as `popd` will work fine, but `cd ..` would need to modified as well with the original approach.
```